lib/src/components/Divider/Divider.tsx

```diff
@@ -0,0 +1,71 @@
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import {
+  dividerContainerHorizontal,
+  dividerContainerVertical,
+  dividerHorizontal,
+  dividerThemeVar,
+  dividerVertical,
+  labelHorizontal,
+  labelVertical,
+} from "./divider.css";
+import { DividerProps } from "./divider.types";
+
+export const Divider = ({
+  type = "solid",
+  label,
+  labelPosition = "center",
+  color = "grey",
+  labelColor = "#000",
+  labelBackground = "#fff",
+  orientation = "horizontal",
+  size = "1px",
+  ...nativeProps
+}: DividerProps) => {
+  const dividerClass =
+    orientation === "horizontal"
+      ? dividerHorizontal({
+          type,
+        })
+      : dividerVertical({
+          type,
+        });
+  const labelClass =
+    orientation === "horizontal"
+      ? labelHorizontal({
+          labelPosition,
+        })
+      : labelVertical({
+          labelPosition,
+        });
+
+  return (
+    <div
+      className={`divider-container ${
+        orientation === "horizontal"
+          ? dividerContainerHorizontal
+          : ` ${dividerContainerVertical}`
+      }`}
+    >
+      <div
+        className={`divider ${dividerClass} `}
+        style={assignInlineVars({
+          [dividerThemeVar.dividerColor]: color,
+          [dividerThemeVar.dividerSize]: size,
+        })}
+        {...nativeProps}
```

Comment:
nativeProps will override className and style. destructure `className` and `style` and merge with styles and classNames of the component (the className on the rightmost wins so take care of that the consumers get maximum customization

Reply:
okay will do that